mongo_orchestration/common.py

```diff
@@ -80,19 +84,36 @@ def _strip_auth(self, proc_params):
         params.pop("clusterAuthMode", None)
         return params
 
-    def mongodb_auth_uri(self, hosts):
-        """Get a connection string with all info necessary to authenticate."""
-        parts = ['mongodb://']
+    def mongodb_uri(self, hosts, uri_opts):
+        """Append URI options for auth, TLS, etc."""
+        """Append TLS options"""
+        if self.ssl_params:
+            ssl_params = self.ssl_params
+            ssl_params.update(DEFAULT_SSL_OPTIONS)
```

Comment:
Ah, I wasn't aware that this modifies the reference. sslParams for example doesn't contain `ssl=true`, it only contains the parameters necessary for the server (see sample orchestration files). `DEFAULT_SSL_OPTIONS` is applied to `self.kwargs` but not `self.ssl_params`. What would be the right way to apply these default options?

Reply:
Oh thanks for explaining that. In this case I don't think we should add any of the `self.ssl_params` to the client URI since those are ssl params for the server itself. I think we should do this:
```python
        if self.ssl_params:  # Server ssl params.
            ssl_params = DEFAULT_SSL_OPTIONS.copy()  # Client ssl params
            ....
```

The above approach would need to add the `tlsInsecure=true` option to match the `ssl_cert_reqs=ssl.CERT_NONE` in DEFAULT_SSL_OPTIONS. 

Alternatively we can keep your current approach but make a copy of self.ssl_params to avoid modifying it:
```python
        if self.ssl_params:
            ssl_params = self.ssl_params.copy()
            ssl_params.update(DEFAULT_SSL_OPTIONS)
```

Reply:
Update to copy the params. `DEFAULT_SSL_OPTIONS` by itself is not enough, as there are additional settings that come into play. The new `SSL_TO_TLS_OPTION_MAPPINGS` variable indicates which of the fields from the configuration's `sslParams` and `DEFAULT_SSL_OPTIONS` map to URI options for SSL. I can add additional tests to ensure we're not adding unknown or wrong URI options if you'd like.

Reply:
SGTM